apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -977,10 +999,10 @@ async function processQuestionServer(phase, codeCaller, data, html, fileData, co
 }
 
 /**
- *
+ * @template {import('./types.js').ExecutionData} T
```

Comment:
Is this a non-generic template?

Reply:
I'm not super familiar using JSDoc-style templates, so I want to save my review for how this is set up for the Typescript version.

Reply:
Or is this like `ExecutionData<T>`

Reply:
So the purpose of this is modeling that all these functions use the same exact type?

Reply:
It's modeling that they use types that are _compatible_ with `ExecutionData`. You can prove to yourself that this matters by removing the constraint from the generic on `processQuestionHtml` and looking at the resulting type errors.

Reply:
I guess my question is will we ever be extending `ExecutionData`? What is the value of the generic over just `ExecutionData`?

Reply:
Try changing it to be `ExecutionData` instead of a generic and check out the resulting type errors 🙂

Reply:
I assume that one of these functions adds additional object keys and I just missed it.

Reply:
It's not about adding extra keys; it's about retaining the _precise_ type that the function is called with. Again, I'd really recommend trying out the change yourself and seeing what happens.